app/services/medium_providers/default.rb

```diff
@@ -26,7 +26,7 @@ def interpolate_vars(pattern)
 
     def unique_id
       @unique_id ||= begin
-        digest = Base64.urlsafe_encode64(Digest::SHA1.digest(medium_uri(entity.operatingsystem.pxedir).to_s), padding: false)
+        digest = Base64.urlsafe_encode64(Digest::SHA1.digest(medium_uri(entity.operatingsystem.try(:pxedir)).to_s), padding: false)
```

Comment:
I wonder if it won't be more complete adding `pxedir` method to `operatingsystem.rb`, so the interface would be a bit more formal.